fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java

```diff
@@ -0,0 +1,848 @@
+        for (long partitionId : partitionIndexMap.rowKeySet()) {
+            Partition partition = table.getPartition(partitionId);
+            long commitVersion = commitVersionMap.get(partitionId);
+            Preconditions.checkState(partition.getNextVersion() == commitVersion,
+                    "partitionNextVersion=" + partition.getNextVersion() + " commitVersion=" + commitVersion);
+            partition.setNextVersion(commitVersion + 1);
+        }
+    }
+
+    @NotNull
+    LakeTable getTableOrThrow(@Nullable LockedDatabase db, long tableId) throws AlterCancelException {
+        if (db == null) {
+            throw new AlterCancelException("Database does not exist");
+        }
+        LakeTable table = db.getTable(tableId);
+        if (table == null) {
+            throw new AlterCancelException("Table does not exist. tableId=" + tableId);
+        }
+        return table;
+    }
+
+    @Override
+    public void replay(AlterJobV2 replayedJob) {
+        LakeTableSchemaChangeJob other = (LakeTableSchemaChangeJob) replayedJob;
+
+        LOG.info("Replaying lake table schema change job. state={} jobId={}", replayedJob.jobState, replayedJob.jobId);
+
+        if (this != other) {
+            Preconditions.checkState(this.type.equals(other.type));
+            Preconditions.checkState(this.jobId == other.jobId);
+            Preconditions.checkState(this.dbId == other.dbId);
+            Preconditions.checkState(this.tableId == other.tableId);
+
+            this.jobState = other.jobState;
+            this.createTimeMs = other.createTimeMs;
+            this.finishedTimeMs = other.finishedTimeMs;
+            this.errMsg = other.errMsg;
+            this.timeoutMs = other.timeoutMs;
+
+            this.partitionIndexTabletMap = other.partitionIndexTabletMap;
+            this.partitionIndexMap = other.partitionIndexMap;
+            this.indexIdMap = other.indexIdMap;
+            this.indexIdToName = other.indexIdToName;
+            this.indexSchemaMap = other.indexSchemaMap;
+            this.indexShortKeyMap = other.indexShortKeyMap;
+            this.hasBfChange = other.hasBfChange;
+            this.bfColumns = other.bfColumns;
+            this.bfFpp = other.bfFpp;
+            this.indexChange = other.indexChange;
+            this.indexes = other.indexes;
+            this.watershedTxnId = other.watershedTxnId;
+            this.storageFormat = other.storageFormat;
+            this.startTime = other.startTime;
+            this.commitVersionMap = other.commitVersionMap;
+            // this.schemaChangeBatchTask = other.schemaChangeBatchTask;
+        }
+
+        try (WriteLockedDatabase db = getWriteLockedDatabase(dbId)) {
+            LakeTable table = (db != null) ? db.getTable(tableId) : null;
+            if (table == null) {
+                return; // do nothing if the table has been dropped.
+            }
+
+            if (jobState == JobState.PENDING) {
```

Comment:
Is this really for lake table? IMO, since we don't have tablet report for lake table, it's ok not add tablet meta to inverted index until the schema change job finished.

Reply:
Better to be consistent with job creation.

Reply:
updated